File: backend/services/execution/algorithms.js
// Verfügbare Algorithmen definieren
export const ALGORITHMS = {
  // Klassifikation
  'RandomForestClassifier': {
    import: 'from sklearn.ensemble import RandomForestClassifier',
    constructor: 'RandomForestClassifier(n_estimators=100, random_state=42)',
    library: 'sklearn',
    base_hyperparameters: {
      n_estimators: 'int',
      random_state: 'int',
    },
    type: 'Classification'
  },
  'SVM': {
    import: 'from sklearn.svm import SVC',
    constructor: 'SVC(random_state=42, probability=True)',
    library: 'sklearn',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Classification'
  },
  'LogisticRegression': {
    import: 'from sklearn.linear_model import LogisticRegression',
    constructor: 'LogisticRegression(random_state=42, max_iter=1000)',
    library: 'sklearn',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Classification'
  },
  'XGBoostClassifier': {
    import: 'from xgboost import XGBClassifier',
    constructor: 'XGBClassifier(random_state=42, eval_metric="logloss")',
    library: 'xgboost',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Classification'
  },
  
  // Regression
  'RandomForestRegressor': {
    import: 'from sklearn.ensemble import RandomForestRegressor',
    constructor: 'RandomForestRegressor(n_estimators=100, random_state=42)',
    library: 'sklearn',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Regression'
  },
  'SVR': {
    import: 'from sklearn.svm import SVR',
    constructor: 'SVR()',
    library: 'sklearn',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Regression'
  },
  'LinearRegression': {
    import: 'from sklearn.linear_model import LinearRegression',
    constructor: 'LinearRegression()',
    library: 'sklearn',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Regression'
  },
  'XGBoostRegressor': {
    import: 'from xgboost import XGBRegressor',
    constructor: 'XGBRegressor(random_state=42)',
    library: 'xgboost',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Regression'
  },
  'NeuralNetworkClassifier': {
    import: 'from torch.nn import MLPClassifier',
    constructor: 'MLPClassifier(random_state=42, max_iter=1000)',
    library: 'pytorch',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Classification'
  },
  'NeuralNetworkRegressor': {
    import: 'from torch.nn import MLPRegressor',
    constructor: 'MLPRegressor(random_state=42, max_iter=1000)',
    library: 'pytorch',
    base_hyperparameters: {
      random_state: 'int',
    },
    type: 'Regression'
  }
};

File: services/mas-service/shared/utils/algorithms.py
"""
Algorithmus-Definitionen mit Standard-Hyperparametern
"""

ALGORITHMS = {
    # Scikit-learn Klassifikation
    'RandomForestClassifier': {
        'library': 'sklearn',
        'hyperparameters': {
            'n_estimators': 100,
            'max_depth': None,
            'min_samples_split': 2,
            'min_samples_leaf': 1,
            'max_features': 'sqrt'
        }
    },
    'LogisticRegression': {
        'library': 'sklearn',
        'hyperparameters': {
            'max_iter': 1000,
            'C': 1.0,
            'solver': 'lbfgs'
        }
    },
    'SVC': {
        'library': 'sklearn',
        'hyperparameters': {
            'C': 1.0,
            'kernel': 'rbf',
            'gamma': 'scale'
        }
    },
    'GradientBoostingClassifier': {
        'library': 'sklearn',
        'hyperparameters': {
            'n_estimators': 100,
            'learning_rate': 0.1,
            'max_depth': 3,
            'min_samples_split': 2
        }
    },
    'KNeighborsClassifier': {
        'library': 'sklearn',
        'hyperparameters': {
            'n_neighbors': 5,
            'weights': 'uniform',
            'algorithm': 'auto'
        }
    },
    'DecisionTreeClassifier': {
        'library': 'sklearn',
        'hyperparameters': {
            'max_depth': None,
            'min_samples_split': 2,
            'min_samples_leaf': 1
        }
    },
    
    # Scikit-learn Regression
    'RandomForestRegressor': {
        'library': 'sklearn',
        'hyperparameters': {
            'n_estimators': 100,
            'max_depth': None,
            'min_samples_split': 2,
            'min_samples_leaf': 1,
            'max_features': 'sqrt'
        }
    },
    'LinearRegression': {
        'library': 'sklearn',
        'hyperparameters': {}
    },
    'Ridge': {
        'library': 'sklearn',
        'hyperparameters': {
            'alpha': 1.0
        }
    },
    'Lasso': {
        'library': 'sklearn',
        'hyperparameters': {
            'alpha': 1.0
        }
    },
    'GradientBoostingRegressor': {
        'library': 'sklearn',
        'hyperparameters': {
            'n_estimators': 100,
            'learning_rate': 0.1,
            'max_depth': 3,
            'min_samples_split': 2
        }
    },
    'SVR': {
        'library': 'sklearn',
        'hyperparameters': {
            'C': 1.0,
            'kernel': 'rbf',
            'gamma': 'scale'
        }
    },
    
    # XGBoost
    'XGBoostClassifier': {
        'library': 'xgboost',
        'hyperparameters': {
            'n_estimators': 100,
            'max_depth': 3,
            'learning_rate': 0.1,
            'subsample': 1.0,
            'colsample_bytree': 1.0
        }
    },
    'XGBoostRegressor': {
        'library': 'xgboost',
        'hyperparameters': {
            'n_estimators': 100,
            'max_depth': 3,
            'learning_rate': 0.1,
            'subsample': 1.0,
            'colsample_bytree': 1.0
        }
    },
    
    # PyTorch Neural Networks
    'NeuralNetwork': {
        'library': 'pytorch',
        'hyperparameters': {
            'nn_hidden_layers': [64, 32],
            'learning_rate': 0.001,
            'batch_size': 16,
            'num_epochs': 10
        }
    }
}

